flink-python/dev/lint-python.sh

```diff
@@ -0,0 +1,508 @@
+    done
+    export PATH 2>/dev/null
+    if [ $? -ne 0 ]; then
+        echo "For some unkown reasons, the py package is not complete,\
+        you should exec the script with the parameter: -f"
+        exit 1
+    fi
+}
+
+# Reset the $PATH
+function deactivate() {
+    # reset old environment variables
+    # ! [ -z ${VAR+_} ] returns true if VAR is declared at all
+    if ! [ -z "${_OLD_PATH+_}" ] ; then
+        PATH="$_OLD_PATH"
+        export PATH
+        unset _OLD_PATH
+    fi
+}
+
+# Tox pytest Check
+function tox_check() {
+    print_function "STAGE" "Stage 2:tox check"
+    # Set created py-env in $PATH for tox's creating virtual env
+    activate
+    $TOX_PATH -c $FLINK_PYTHON_DIR/tox.ini --recreate 2>&1 | tee -a $LOG_FILE
+
+    TOX_RESULT=$((grep -c "congratulations :)" "$LOG_FILE") 2>&1)
+    if [ $TOX_RESULT -eq '0' ]; then
+        print_function "STAGE" "tox checked ... [FAILED]"
+    else
+        print_function "STAGE" "tox checked ... [SUCCESS]"
+    fi
+    # Reset the $PATH
+    deactivate
+    # If check failed, stop the running script.
+    if [ $TOX_RESULT -eq '0' ]; then
+        exit 1
+    fi
+    STAGE="flake8"
+    STEP="0"
+}
+
+# Flake8 codestyle check
+function flake8_check {
+    local PYTHON_SOURCE="$(find . \( -path ./dev -o -path ./.tox \) -prune -o -type f -name "*.py" -print )"
+
+    print_function "STAGE" "Stage 3:flake8 codestyle check"
+    if [ ! -f "$FLAKE8_PATH" ]; then
+        echo "For some unkown reasons, the flake8 package is not complete,\
+        you should exec the script with the parameter: -f"
+    fi
+
+    if [[ ! "$PYTHON_SOURCE" ]]; then
+        echo "No python files found!  Something is wrong exiting."
+        exit 1;
+    fi
+
+    ($FLAKE8_PATH  --config=tox.ini $PYTHON_SOURCE) 2>&1 | tee -a $LOG_FILE
+
+    PYCODESTYLE_STATUS=$?
+    if [ $PYCODESTYLE_STATUS -ne 0 ]; then
+        print_function "STAGE" "pycodestyle checks ... [FAILED]"
+    else
+        print_function "STAGE" "pycodestyle checks ... [SUCCESS]"
+    fi
+}
+
+# CURRENT_DIR is "flink/flink-python/dev/"
+CURRENT_DIR="$(cd "$( dirname "$0" )" && pwd)"
+
+# FLINK_PYTHON_DIR is "flink/flink-python"
+FLINK_PYTHON_DIR=$(dirname "$CURRENT_DIR")
+pushd "$FLINK_PYTHON_DIR" &> /dev/null
+
+# conda path
+CONDA_PATH=$CURRENT_DIR/.conda/bin/conda
+
+# tox path
+TOX_PATH=$CURRENT_DIR/.conda/bin/tox
+
+# flake8 path
+FLAKE8_PATH=$CURRENT_DIR/.conda/bin/flake8
+
+_OLD_PATH="$PATH"
+
+SUPPORT_OS=("Darwin" "Linux")
+
+# the file stores the success progress.
+STAGE_FILE=$CURRENT_DIR/.stage.txt
+
+# the dir includes all kinds of py env installed.
+VIRTUAL_ENV=$CURRENT_DIR/.conda/envs
+
+LOG_DIR=$CURRENT_DIR/log
+
+if [ "$FLINK_IDENT_STRING" == "" ]; then
+    FLINK_IDENT_STRING="$USER"
+fi
+if [ "$HOSTNAME" == "" ]; then
+    HOSTNAME="$HOST"
+fi
+
+# the log file stores the checking result.
+LOG_FILE=$LOG_DIR/flink-$FLINK_IDENT_STRING-python-$HOSTNAME.log
+create_dir $LOG_DIR
+if [ $? -ne 0 ]; then
```

Comment:
create_dir will make the process exit if create file failed. So the check here seems unnecessary. What do you think?